nmdc_runtime/api/analytics.py

```diff
@@ -0,0 +1,64 @@
+"""
+Based on <https://github.com/tom-draper/api-analytics/tree/main/analytics/python/fastapi>
+under MIT License <https://github.com/tom-draper/api-analytics/blob/main/analytics/python/fastapi/LICENSE>
+"""
+
+from datetime import datetime
+import threading
+from time import time
+from typing import Dict, List
+
+from starlette.middleware.base import BaseHTTPMiddleware, RequestResponseEndpoint
+from starlette.requests import Request
+from starlette.responses import Response
+from starlette.types import ASGIApp
+from toolz import merge
+
+from nmdc_runtime.api.db.mongo import get_mongo_db
+
+_requests = []
+_last_posted = datetime.now()
+
+
+def _post_requests(collection: str, requests_data: List[Dict], source: str):
```

Comment:
Suggestion: Instead of "post", maybe "save" or "write". I don't think this warrants blocking a merge, though.